File: MarketJava/src/Utils/Pair.java
package Utils;

/**
 * Simple pair container to store two objects
 * inspiration taken from class provided in C++ STL & package javafx
 * did not want to use javafx as unknown if this is on lab computers
 *
 * @param <T>
 *     Any type of object
 * @param <S>
 *     Any type of object
 */
public class Pair<T, S>{
    T t;
    S s;
    public Pair(T t, S s)
    {
        this.t = t;
        this.s = s;
    }

    public T first()
    {
        return this.t;
    }

    public S second()
    {
        return this.s;
    }

}

File: MarketCSharp/Utils/Pair.cs
using System;
using System.Collections.Generic;
using System.Text;
/**
 * Pair's purpose is as a container class taking two generic types as arguments
 * with getter functions.
 * 
 * Inspiration/Reference for class from c++ STL
 */
namespace Utils
{
    public class Pair<T, U>
    {
        private T _first;
        private U _second;
        public T First { get { return _first; } }
        public U Second {  get { return _second; } }

        public Pair(T first, U second)
        {
            _first = first;
            _second = second;
        }
    }
}
